prow/tide/tide.go

```diff
@@ -766,23 +766,42 @@ func unsuccessfulContexts(contexts []Context, cc contextChecker, log *logrus.Ent
 	return failed
 }
 
-func pickSmallestPassingNumber(log *logrus.Entry, ghc githubClient, prs []PullRequest, cc map[int]contextChecker) (bool, PullRequest) {
+func hasAllLabels(pr PullRequest, labels []string) bool {
+	if len(labels) == 0 {
+		return true
+	}
+	prLabels := sets.NewString()
+	for _, l := range pr.Labels.Nodes {
+		prLabels.Insert(string(l.Name))
+	}
+	return len(prLabels.Intersection(sets.NewString(labels...))) == len(labels)
+}
+
+func pickSmallestPassingNumber(log *logrus.Entry, ghc githubClient, prs []PullRequest, cc map[int]contextChecker, isPassingTestsFunc func(*logrus.Entry, githubClient, PullRequest, contextChecker) bool, priorities []config.TidePriority) (bool, PullRequest) {
```

Comment:
Could you rename this func to something like `pickHighestPriorityPR` or sth like that to account for the changed behavior?